platform/view/services/storage/vault/vaultstore_test.go

```diff
@@ -26,6 +30,133 @@ const (
 	unknown
 )
 
+type matrixItem struct {
+	pagination driver.Pagination
+	matcher    []types.GomegaMatcher
+}
+
+var matrix = []matrixItem{
+	{
+		pagination: common.NewNoPagination(),
+		matcher: []types.GomegaMatcher{
+			gomega.ConsistOf(
+				gomega.HaveField("TxID", gomega.Equal("txid1")),
+				gomega.HaveField("TxID", gomega.Equal("txid2")),
+				gomega.HaveField("TxID", gomega.Equal("txid10")),
+				gomega.HaveField("TxID", gomega.Equal("txid12")),
+				gomega.HaveField("TxID", gomega.Equal("txid21")),
+				gomega.HaveField("TxID", gomega.Equal("txid100")),
+				gomega.HaveField("TxID", gomega.Equal("txid200")),
+				gomega.HaveField("TxID", gomega.Equal("txid1025")),
+			),
+		},
+	},
+	{
+		pagination: NewOffsetPagination(0, 2),
+		matcher: []types.GomegaMatcher{
+			gomega.ConsistOf(
+				gomega.HaveField("TxID", gomega.Equal("txid1")),
+				gomega.HaveField("TxID", gomega.Equal("txid2")),
+			),
+			gomega.ConsistOf(
+				gomega.HaveField("TxID", gomega.Equal("txid10")),
+				gomega.HaveField("TxID", gomega.Equal("txid12")),
+			),
+			gomega.ConsistOf(
+				gomega.HaveField("TxID", gomega.Equal("txid21")),
+				gomega.HaveField("TxID", gomega.Equal("txid100")),
+			),
+			gomega.ConsistOf(
+				gomega.HaveField("TxID", gomega.Equal("txid200")),
+				gomega.HaveField("TxID", gomega.Equal("txid1025")),
+			),
+		},
+	},
+}
+
+func NewOffsetPagination(offset int, pageSize int) *common.OffsetPagination {
+	offsetPagination, err := common.NewOffsetPagination(offset, pageSize)
+	if err != nil {
+		assert1.NoError(err)
+	}
+	return offsetPagination
+}
+
+func getAllTxStatuses(store driver.VaultStore, pagination driver.Pagination) ([]driver.TxStatus, error) {
+	p, err := store.GetAllTxStatuses(context.Background(), pagination)
+	if err != nil {
+		return nil, err
+	}
+	statuses, err := collections.ReadAll(p.Items)
+	if err != nil {
+		return nil, err
+	}
+	return statuses, nil
+}
+
+func testPagination(t *testing.T, store driver.VaultStore) {
+	g := gomega.NewWithT(t)
+	err := store.SetStatuses(context.Background(), driver.TxStatusCode(valid), "",
+		"txid1", "txid2", "txid10", "txid12", "txid21", "txid100", "txid200", "txid1025")
+	assert.NoError(t, err)
+
+	for _, item := range matrix {
+		page := 0
+		for {
+			statuses, err := getAllTxStatuses(store, item.pagination)
+			assert.NoError(t, err)
+			if len(statuses) == 0 {
+				break
+			}
+			g.Expect(statuses).To(item.matcher[page])
+			item.pagination, err = item.pagination.Next()
+			assert.NoError(t, err)
+			page++
+		}
+		for {
+			item.pagination, err = item.pagination.Prev()
+			assert.NoError(t, err)
```

Comment:
Don't mix different assertion strategies. You currently use gomega, assert and assert1. Remove assert and assert1 and only stay with gomega.

```
Expect(err).ToNot(HaveOccurred())
```